modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/fid/IndexedFidWriter.java

```diff
@@ -206,11 +206,12 @@ private void finishLastWrite() throws IOException {
      * Increments the fidIndex by 1. Indicates that a feature was removed from the location. This is
      * intended to ensure that FIDs stay constant over time. Consider the following case of 5
      * features. feature 1 has fid typename.0 feature 2 has fid typename.1 feature 3 has fid
-     * typename.2 feature 4 has fid typename.3 feature 5 has fid typename.4 when feature 3 is
-     * removed/deleted the following usage of the write should take place: next(); (move to feature
-     * 1) next(); (move to feature 2) next(); (move to feature 3) remove();(delete feature 3)
-     * next(); (move to feature 4) // optional write(); (write feature 4) next(); (move to feature
-     * 5) write(); (write(feature 5)
+     * typename.2 feature 4 has fid typename.3 feature 5 has fid typename.4
```

Comment:
This is indeed hard to follow and partly outdated. I can see it updating the "removed" counter, not fidIndex. 
Probably a victim of mass reformat, do you have time to turn it into a bullet list, ``<ul><li>...</li></ul>``, structure?
If not, no problem, it's not the objective of the fix.